### PR TITLE
feat: [SSCA-1144]: Adding permissions to Exemptions API and iterator to expire exemptions

### PR DESCRIPTION
**Pull request by dhruv-harness**
## Describe your changes
Adding the RBAC permissions to Exemptions APIs.
Adding iterator to expire old approved exemptions.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57241)

### CODE REVIEW

**Comment by dhruv-harness** [comment]
trigger uts

File: ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java [hunk 1]
```diff
@@ -82,19 +97,22 @@ public Response listExemptionsForProject(String org, String project, String harn
   }
 
   @Override
+  @NGAccessControlCheck(resourceType = SSCA_ENFORCEMENT_EXEMPTION, permission = REVIEW_PERMISSION)
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java:100]: You haven't added @AccountIdentifier @OrgIdentifier @ProjectIdentifier @ResourceIdentifier to org, project, account. The RBAC will not work.

File: ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java [hunk 2]
```diff
@@ -30,8 +31,16 @@
 @OwnedBy(HarnessTeam.SSCA)
 @NextGenManagerAuth
 public class ExemptionsApiImpl implements ExemptionsApi {
+  private static final String SSCA_ENFORCEMENT_EXEMPTION = "SSCA_ENFORCEMENT_EXEMPTION";
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java:34]: Please move these to a common class where we have Remediation permissions defined as well

**Review by reetika04 — CHANGES_REQUESTED** [review]

File: ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java (hunk 2 above)
Comment (dhruv-harness) [reply ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java]: Sure

File: ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java (hunk 1 above)
Comment (dhruv-harness) [reply ssca-manager/service/src/main/java/io/harness/ssca/api/ExemptionsApiImpl.java]: Adding

**Comment by dhruv-harness** [comment]
trigger uts

**Comment by dhruv-harness** [comment]
trigger uts

**Review by reetika04 — APPROVED** [review]

**Pull request merged**